// @nivo/legends/lib/props.js
'use strict';

exports.__esModule = true;
exports.LegendPropShape = undefined;

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _constants = require('./constants');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * The prop type is exported as a simple object instead of `PropTypes.shape`
 * to be able to add extra properties.
 *
 * @example
 * ```javascript
 * import { LegendPropShape } from '@nivo/legends'
 *
 * const customLegendPropType = PropTypes.shape({
 *     ...LegendPropShape,
 *     extra: PropTypes.any.isRequired,
 * })
 * ```
 */
/*
 * This file is part of the nivo project.
 *
 * Copyright 2016-present, Raphaël Benitte.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */
var LegendPropShape = exports.LegendPropShape = {
    data: _propTypes2.default.arrayOf(_propTypes2.default.shape({
        label: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.number]).isRequired,
        fill: _propTypes2.default.string.isRequired
    })),

    // position & layout
    anchor: _propTypes2.default.oneOf([_constants.ANCHOR_TOP, _constants.ANCHOR_TOP_RIGHT, _constants.ANCHOR_RIGHT, _constants.ANCHOR_BOTTOM_RIGHT, _constants.ANCHOR_BOTTOM, _constants.ANCHOR_BOTTOM_LEFT, _constants.ANCHOR_LEFT, _constants.ANCHOR_TOP_LEFT, _constants.ANCHOR_CENTER]).isRequired,
    translateX: _propTypes2.default.number,
    translateY: _propTypes2.default.number,
    direction: _propTypes2.default.oneOf([_constants.DIRECTION_ROW, _constants.DIRECTION_COLUMN]).isRequired,

    // items
    itemWidth: _propTypes2.default.number.isRequired,
    itemHeight: _propTypes2.default.number.isRequired,
    itemDirection: _propTypes2.default.oneOf([_constants.DIRECTION_LEFT_TO_RIGHT, _constants.DIRECTION_RIGHT_TO_LEFT, _constants.DIRECTION_TOP_TO_BOTTOM, _constants.DIRECTION_BOTTOM_TO_TOP]),
    itemsSpacing: _propTypes2.default.number,
    symbolSize: _propTypes2.default.number,
    symbolSpacing: _propTypes2.default.number,
    symbolShape: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.func])
};
